Library/Formula/boost.rb

```diff
@@ -36,6 +36,14 @@ class Boost < Formula
     cause "Dropped arguments to functions when linking with boost"
   end
 
+  # Patch for odeint. Can be removed with the next release of Boost (1.58).
+  # See the issue here:
+  # https://github.com/headmyshoulder/odeint-v2/issues/138
+  # The fix has been merged into the Boost project here:
+  # https://github.com/boostorg/odeint/commit/0ca187cd6e83ddbaa5eedaa5d0d570986f351263
+  # It is needed to compile OMPL (http://ompl.kavrakilab.org/).
+  patch :DATA
```

Comment:
Looks like you should be able to use `https://github.com/boostorg/odeint/commit/0ca187.diff`, perhaps with `p1` or `p2`?